css/BEM-style.md

```diff
@@ -0,0 +1,223 @@
+
+### Objectives
+
+**Participants will be able to:**
+- Understand what BEM naming architecture is and why it is used.
+- Understand how it helps with writing a scalable and maintanable CSS file.
+- Use BEM naming architecture on a small example project.
+
+### Materials
+
+The best reference/resource to get you started: 
+- [BEM documentaion](http://getbem.com/introduction/) (You can understand the basics of BEM -2 min read). 
+  
+
+### Lessons
+
+- To understand how BEM is used for styling elements, check out this great article on Mr-Frontend Blog: ['How to write better CSS with BEM'](https://mrfrontend.org/2017/10/write-better-css-with-bem/) by Raymon S (10 min read).
+
+
+### Common Mistakes / Misconceptions
+
+- **"My CSS styles never have conflicts since I utilize ID's most often to style my elements"**   
+Using ID's can definitely help you avoid any kind of specificity conflicts, since they have a priority over classes and tags, but that would mean, your styles are not reusable and you don't follow the DRY( Dont'Repeat Yourself) principle. New styles need to be written for every element even if some of them share a few common feature (Eg: buttons on a page, navlinks etc).
+
+### Guided Practice
+We'll make use of the BEM method to create few buttons and a hero-box .
+
+> Note: You must have prior knowledge of CSS and HTML to understand what BEM is and how to make use of it.
+
+**1.Understanding what BEM is?:** 
+
+BEM is a naming methodology/principle which stand for "Block", "Element", "Modifier" to describe and differentiate, the different components of your webpage in a way that is easy to understand,maintain and scale as and when required.
+In BEM methodology the important point to note is that each and every HTML element will be accessed using a class only, we don't use ID's or tags to style them, to keep specificity uniform across the entire project i.e. (0,1,0) : (ID's, Classes, Tags).
+
+- The **'Block'** are the standalone components in a webpage that are meaningful on its own.
+ 
+  >Example includes: navbar,header,footer,hero-boxes, buttons etc.
+
+-  The **'Element'** are the components that have no standalone meaning, but when tied to a block, represents a part of the block. It expresses a semantic meaning only when tied to a block.  
+   
+   >Example includes: nav-links,footer-links, headings and sub headings inside heroboxes etc.
+
+-  The **'Modifier'** are like flags that differentiate blocks or elements inside block in terms of its appreance.
+   
+   >Example includes: buttons which are muted, navlinks which are active, disabled etc. 
+
+
+**2.Understanding how BEM is written (The Syntax) ?:** 
+
+To understand this, let us consider that we are making two different buttons: one which is in active state and one which is in ghosted state.
+Then the HTML makrup with the classes will be:
+```
+<a href="#"class="btn btn--active">I know!</a>
```

Comment:
These aren't buttons, they're links.

Reply:
I agree, please call them links.

Reply:
The reason why I named them and placed under the buttons group was that they were styled to look like buttons. And that's all. But now I have corrected it.

Reply:
This brings up a common mistake so it would be useful to clarify that regardless of styling (which is more of a Designer concern), which HTML tag is the appropriate one to use in different circumstances.

Reply:
Yeah, I noticed how it could be confusing, so I have changed as per requirements and also explained in more detail whenever it could have a possible state of confusion.